src/components/Checkbox/Checkbox.js

```diff
@@ -134,8 +201,12 @@ const Checkbox = React.forwardRef(function Checkbox(props, ref) {
     setState(newState);
     if (onChange) onChange(e);
   };
-  const CheckboxComponent =
-    variant === 'flat' ? StyledFlatCheckbox : StyledCheckbox;
+
+  const CheckboxComponent = {
```

Comment:
What do you think about declaring this `CheckboxComponent` outside of render? Since the components are static I don't see a need to declare them on every render, what do you think?

Just to be clear, my idea is to declare the checkbox components outside of render and only instantiate the component that'll be used inside of render, like:

```javascript
// outside of render
const CheckboxComponents = {
  flat: StyledFlatCheckbox,
  default: StyledCheckbox,
  menu: StyledMenuCheckbox
}

// inside of render
const CheckboxVariant = CheckboxComponents[variant]
```

Does that make sense for you?